poc/suite_bls12381g1.sage

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/sage
+# vim: syntax=python
+
+import hashlib
+import sys
+try:
+    from sagelib.common import sgn0_be
+    from sagelib.h2c_suite import BasicH2CSuiteDef, BasicH2CSuite, IsoH2CSuiteDef, IsoH2CSuite
+    from sagelib.svdw_generic import GenericSvdW
+    from sagelib.sswu_generic import GenericSSWU
+    from sagelib.suite_p256 import _test_suite
+    from sagelib.iso_values import iso_bls12381g1
+except ImportError:
+    sys.exit("Error loading preprocessed sage files. Try running `make clean pyfiles`")
+
+p = 0x1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaaab
+F = GF(p)
+A = F(0)
+B = F(4)
+Ap = F(0x144698a3b8e9433d693a02c96d4982b0ea985383ee66a8d8e8981aefd881ac98936f8da0e0f97f5cf428082d584c1d)
+Bp = F(0x12e2908d11688030018b12e8753eee3b2016c1f0f24f4070a0b9c14fcef35ef55a23215a316ceaa5d1cc48e98e172be0)
+h_eff = 0xd201000000010001
+iso_map = iso_bls12381g1()
+
+bls12381g1_svdw_def = BasicH2CSuiteDef(F, A, B, sgn0_be, hashlib.sha256, 64, GenericSvdW, h_eff, True, "asdf")
```

Comment:
How does L=64 was calculated?

Reply:
That's the value from the suite; it's calculated using the procedure described in the hash_to_base section. p = 381 bits, the curve is ~128-bit security; 381 + 128 = 509. Ceil(509 / 8) = 64.